azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/state/WorkItemTypeStateService.java

```diff
@@ -0,0 +1,77 @@
+ * under the License.
+ */
+package com.synopsys.integration.azure.boards.common.service.state;
+
+import java.lang.reflect.Type;
+
+import com.google.gson.reflect.TypeToken;
+import com.synopsys.integration.azure.boards.common.http.AzureHttpService;
+import com.synopsys.integration.azure.boards.common.http.HttpServiceException;
+import com.synopsys.integration.azure.boards.common.model.AzureArrayResponseModel;
+import com.synopsys.integration.azure.boards.common.util.AzureSpecTemplate;
+
+public class WorkItemTypeStateService {
+    public static final AzureSpecTemplate API_SPEC_ORGANIZATION_PROJECT_WORKITEMTYPE_STATES = new AzureSpecTemplate("/{organization}/{project}/_apis/wit/workitemtypes/{type}/states");
+    public static final AzureSpecTemplate API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPE_STATES = new AzureSpecTemplate("/{organization}/_apis/work/processes/{processId}/workItemTypes/{witRefName}/states");
+    public static final AzureSpecTemplate API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPE_STATE_INDIVIDUAL = new AzureSpecTemplate("/{organization}/_apis/work/processes/{processId}/workItemTypes/{witRefName}/states/{stateId}");
+
+    private final AzureHttpService azureHttpService;
+
+    public WorkItemTypeStateService(AzureHttpService azureHttpService) {
+        this.azureHttpService = azureHttpService;
+    }
+
+    public AzureArrayResponseModel<WorkItemTypeStateResponseModel> getStatesForProject(String organizationName, String projectIdOrName, String workItemType) throws HttpServiceException {
+        String requestSpec = API_SPEC_ORGANIZATION_PROJECT_WORKITEMTYPE_STATES
+                                 .defineReplacement("{organization}", organizationName)
+                                 .defineReplacement("{project}", projectIdOrName)
+                                 .defineReplacement("{type}", workItemType)
+                                 .populateSpec();
+        requestSpec = String.format("%s?%s=%s", requestSpec, AzureHttpService.AZURE_API_VERSION_QUERY_PARAM_NAME, "5.1-preview.1");
+        Type responseType = new TypeToken<AzureArrayResponseModel<WorkItemTypeStateResponseModel>>() {}.getType();
+        return azureHttpService.get(requestSpec, responseType);
+    }
+
+    public AzureArrayResponseModel<WorkItemTypeProcessStateResponseModel> getStatesForProcess(String organizationName, String processId, String workItemTypeRefName) throws HttpServiceException {
+        String requestSpec = API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPE_STATES
+                                 .defineReplacement("{organization}", organizationName)
+                                 .defineReplacement("{processId}", processId)
+                                 .defineReplacement("{witRefName}", workItemTypeRefName)
+                                 .populateSpec();
+        requestSpec = String.format("%s?%s=%s", requestSpec, AzureHttpService.AZURE_API_VERSION_QUERY_PARAM_NAME, "5.1-preview.1");
```

Comment:
Are you sure we should be using a version type of *-preview?

Reply:
There are several pieces of functionality that require using preview versions. Luckily these are not subject to change, but they may not be fully tested.